server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -197,7 +197,27 @@ class LivyServer extends Logging {
           livyConf.get(AUTH_KERBEROS_NAME_RULES))
         server.context.addFilter(holder, "/*", EnumSet.allOf(classOf[DispatcherType]))
         info(s"SPNEGO auth enabled (principal = $principal)")
-
+      case authType@LdapAuthenticationHandler.TYPE =>
+        val holder = new FilterHolder(new AuthenticationFilter())
+        holder.setInitParameter(AuthenticationFilter.AUTH_TYPE, authType)
+        val url = livyConf.get(LivyConf.AUTH_LDAP_URL)
+        if (url != null) {
+          holder.setInitParameter(LdapAuthenticationHandler.PROVIDER_URL, url)
+        }
+        val baseDN = livyConf.get(LivyConf.AUTH_LDAP_BASE_DN)
+        if (baseDN != null) {
+          holder.setInitParameter(LdapAuthenticationHandler.BASE_DN, baseDN)
+        }
+        val domain = livyConf.get(LivyConf.AUTH_LDAP_USERNAME_DOMAIN)
+        if (domain != null) {
+          holder.setInitParameter(LdapAuthenticationHandler.LDAP_BIND_DOMAIN, domain)
+        }
```

Comment:
Does `url`, `domain` and `baseDN` are all required to enable LDAP, or either of them can be null?

Also the code can be simplified as below, no need to do null check like Java.

```
Option(livyConf.get(LivyConf.AUTH_LDAP_USERNAME_DOMAIN)).foreach(holder.setInitParameter(LdapAuthenticationHandler.LDAP_BIND_DOMAIN, _))
```